src/app/shared/components/filters/filters.component.ts

```diff
@@ -50,11 +53,17 @@ export class FiltersComponent {
     return FilterType;
   }
 
-  public get isFiltered() {
+  public get isFilterChangedSinceLastApply() {
+    return !_.isEqual(this._appliedFilters, this._filters);
+  }
+
+  public get isFilterNotEmpty() {
     return !!this._filters.find(f => f.value);
   }
 
   public apply() {
+    this._appliedFilters = _.cloneDeep(this.filters);
```

Comment:
I guess internally we should use original property that faced behind getter / setter. In this example, it should be `this._filters`.